### PR TITLE
Update Keyvault sdk from t1 to t2

### PR DESCRIPTION
**Pull request by Tianyun-G**
1. Update Keyvault and Identity sdk from t1 to t2

2. Update files:

- i. KeyVaultAuthSample.cs
- ii. key-vault-dotnet-authentication.csproj

@scottaddie,@AlexGhiondea,@schaabs,@christothes and @heaths for notification.

### CODE REVIEW

**Comment by v-xuto** [comment]
@scottaddie,@AlexGhiondea,@schaabs,@christothes, @heaths  Could you please review this PR?

**Comment by ahsonkhan** [comment]
I don't work on the .NET KeyVault SDKs, so I don't think I am the appropriate person to review this. Maybe it was meant for someone else :)

For C++ SDK, we have @vhvb1989 and @gearama.

**Comment by vhvb1989** [comment]
> I don't work on the .NET KeyVault SDKs, so I don't think I am the appropriate person to review this. Maybe it was meant for someone else :)
> 
> For C++ SDK, we have @vhvb1989 and @gearama.

I think they were just trying to tag @antkmsft but ended up tagging you instead =P . This looks to be related to Identity

**Comment by AlexGhiondea** [comment]
@heaths can you take a look?

File: KeyVaultAuthSample.cs [hunk 1]
```diff
@@ -60,26 +55,25 @@ private async Task authUsingADALCallbackAsync(string vaultBaseURL)
...
-            // Set up a KV Client with an ADAL authentication callback function
-            KeyVaultClient kvClient = new KeyVaultClient(
-                async (string authority, string resource, string scope) =>
+            var credential = new DefaultAzureCredential();
+            SecretClient secretClient = new SecretClient(new Uri(vaultBaseURL), credential);
+
+            // Set and get an example secret
+            await secretClient.SetSecretAsync("test-secret", "test-secret-value-using-adal");
+
+            await foreach (SecretProperties secret in secretClient.GetPropertiesOfSecretsAsync())
```
Comment (heaths) [inline KeyVaultAuthSample.cs]: Is there some reason you're getting all secrets? This is against best practices of getting only what you need, if you know up front. This is a practice we've been trying to get customers away from with newer samples.

**Review by heaths — CHANGES_REQUESTED** [review]

**Comment by Tianyun-G** [comment]
@ahsonkhan - I'm very sorry to add you. I've already removed you.

File: KeyVaultAuthSample.cs (hunk 1 above)
Comment (Tianyun-G) [reply KeyVaultAuthSample.cs]: @heaths - I have updated, please review.

**Review by heaths — APPROVED** [review]

**Comment by v-xuto** [comment]
@heaths  Do you have permission for this repo? If so, please help us merge this PR.

**Comment by heaths** [comment]
I don't.

**Comment by v-xuto** [comment]
@Lusitanian  Do you have permission for this repo? If so, please help us merge this PR.

**Comment by daviddesberg** [comment]
@v-xuto unfortunately it doesn't seem that I have write access to this repository anymore

**Comment by v-xuto** [comment]
@Lusitanian Okay, I'm sorry to disturb you.

**Comment by daviddesberg** [comment]
@v-xuto no worries - see offline chat. will see if we can figure out who owns this now

**Pull request merged**